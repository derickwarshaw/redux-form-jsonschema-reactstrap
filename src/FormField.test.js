import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { FormField } from './index';
import { omit } from 'lodash';

const schema = {
  title: 'Foo',
  description: 'foobar',
  type: 'number',
  meta: {
    vis: {
      ordinal: 0,
      editable: true,
      component: 'NumberInputField'
    }
  }
};

const schemaNoDescription = omit(schema, ['description']);

const testFormFieldSnapshot = (options, testRenderer = renderer) => () => {
  const tree = testRenderer
    .create(
      <FormField {...options}>
        <input type="number" />
      </FormField>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
};

const testShallowFormFieldSnapshot = options => () => {
  const wrapper = shallow(
    <FormField {...options}>
      <input type="number" />
    </FormField>
  );

  expect(wrapper.getNodes()).toMatchSnapshot();
}

describe('Render FormField', () => {
  it(
    'should render',
    testShallowFormFieldSnapshot({
      name: 'foo',
      schema,
      required: true,
      meta: { touched: true, error: 'error', warning: 'warning' }
    })
  );

  it(
    'should render without tooltip',
    testFormFieldSnapshot({
      name: 'foo',
      schema: schemaNoDescription,
      required: true,
      meta: { touched: true, error: 'error', warning: 'warning' }
    })
  );
});
