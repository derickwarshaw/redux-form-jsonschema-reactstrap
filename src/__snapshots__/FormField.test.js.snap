// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FormField should render 1`] = `
<div>
  <label
    className=""
    htmlFor="foo"
    id="foo-label"
  >
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
    Foo
  </label>
  <div
    id="foo-error"
  >
    error
  </div>
  <div
    id="foo-warning"
  >
    warning
  </div>
</div>
`;
